core/server/api/canary/members.js

```diff
@@ -761,7 +761,7 @@ module.exports = {
                             if (membersWithStripeCustomers.length) {
                                 await Promise.map(membersWithStripeCustomers, async (stripeMember) => {
                                     try {
-                                        await membersService.api.members.linkStripeCustomer(stripeMember.stripe_customer_id, stripeMember);
+                                        await membersService.api.members.linkStripeCustomerById(stripeMember.stripe_customer_id, stripeMember.id);
```

Comment:
Something I don't have strong opinion about but wanted to know what you think. Did you consider  keeping existing `linkStripeCustomer` methods as is without renaming to `ById`? Instead check for type of second argument that's passed in - if integer would fetch the records, if model do the normal logic.

I think the `ById` approach is very explicit and gives a signal about possible things that might be going on inside the method call, like fetching the model. On the other side, it expands the API surface :thinking:

Reply:
I did consider that initially yeah, the reason I decided against it is twofold:
1) Like you say this is more explicit, and I'm finding as this codebase grows, explicitness is becoming more valuable
2) I'm hoping to "type" the `members-api` code in the future, and JSDoc typings with polymorphic functions is quite tricky!